lib/segment/src/index/field_index/map_index/immutable_map_index.rs

```diff
@@ -16,8 +17,11 @@ use crate::common::rocksdb_wrapper::DatabaseColumnWrapper;
 use crate::index::field_index::immutable_point_to_values::ImmutablePointToValues;
 
 pub struct ImmutableMapIndex<N: MapIndexKey + ?Sized> {
-    value_to_points: HashMap<N::Owned, Range<u32>>,
+    value_to_points: HashMap<N::Owned, (Range<u32>, u32)>,
```

Comment:
Maybe it's worth converting `(Range<u32>, u32)` into a struct with named fields.

Reply:
Agreed. Done!